src/layerstack.stanza

```diff
@@ -20,6 +20,12 @@ defstruct LayerMaterial <: Hashable:
 
 defmulti create-material (x:LayerMaterial) -> JITXDef
 
+defmethod equal? (ma:LayerMaterial, mb:LayerMaterial) :
```

Comment:
The `JITXDef` types are `Equalable` - so I think you should be able to just add `equalable => true` to the `defstruct LayerMaterial` definition.  For example: 

```
public-when(TESTING) defstruct LayerMaterial <: Hashable & Equalable :
  name:Maybe<String>
  type:MaterialType
  description:Maybe<String>
  material-def:Maybe<JITXDef> with: (
    setter => set-material-def
  )
with:
  equalable => true
  ; If we want to make this public - then we also need this.
  hashable => true
```

Further, you are missing the `description` so this `equal?` method will not work in all cases. 

We should write a test for this to be sure.

Reply:
I don't think that is true.

Reply:
Adding just `true => equalable` works.

This `equal?` function is already tested since I need it to test `state[0]` etc. For example,
```
deftest(layerstack) test-get :
  #EXPECT(test-stack[0] == copper-3oz)
  #EXPECT(test-stack[1] == core-FR4-2)
  #EXPECT(test-stack[2] == copper-2oz)
  #EXPECT(test-stack[3] == prepreg-FR4)
  ```